### PR TITLE
fix: upgrade ts types to latest version

### PR DESCRIPTION
**Pull request by Trancever**
### Summary

This PR upgrades react native types to v0.63 and fixes few eslint warnings.

### Test plan

Reinstall node_modules and run `yarn typescript`

### CODE REVIEW

File: src/components/Appbar/Appbar.tsx [hunk 1]
```diff
@@ -94,7 +94,8 @@ const Appbar = ({ children, dark, style, theme, ...rest }: Props) => {
     isDark =
       backgroundColor === 'transparent'
         ? false
-        : !color(backgroundColor).isLight();
+        : // @ts-ignore
```
Comment (satya164) [inline src/components/Appbar/Appbar.tsx:97]: Why is this needed? Can `backgroundColor` be non-string? Then we should check if it's a string first. (Same in other files)

File: src/components/BottomNavigation.tsx [hunk 2]
```diff
@@ -749,6 +750,7 @@ const BottomNavigation = ({
                 onPress: () => handleTabPress(index),
                 testID: getTestID({ route }),
                 accessibilityLabel: getAccessibilityLabel({ route }),
+                // @ts-ignore
```
Comment (satya164) [inline src/components/BottomNavigation.tsx:753]: Add a comment too

**Review by satya164 — CHANGES_REQUESTED** [review]

File: src/components/ActivityIndicator.tsx [hunk 3]
```diff
@@ -146,7 +146,7 @@ const ActivityIndicator = ({
   const containerStyle = {
     width: size,
     height: size / 2,
-    overflow: 'hidden',
+    overflow: 'hidden' as ViewStyle['overflow'],
```
[the remark this answers is not in the document]
Comment (Trancever) [reply src/components/ActivityIndicator.tsx:149]: 👍

File: src/components/Appbar/Appbar.tsx (hunk 1 above)
Comment (maltoze) [reply src/components/Appbar/Appbar.tsx:97]: Colors simply are no longer just strings in v0.63.

File: src/components/Button.tsx [hunk 4]
```diff
@@ -265,6 +265,7 @@ const Button = ({
         onPressIn={handlePressIn}
         onPressOut={handlePressOut}
         accessibilityLabel={accessibilityLabel}
+        // @ts-ignore We keep old a11y props for backwards compat with old RN versions
         accessibilityTraits={disabled ? ['button', 'disabled'] : 'button'}
         accessibilityComponentType="button"
```
Comment (imnotjames) [inline src/components/Button.tsx:270]: I think `accessibilityComponentType` also needs to be ignored - unless I'm misunderstanding how `// @ts-ignore` works

**Comment by chybisov** [comment]
Do you guys have any plans to merge this? It's very annoying to see types errors 😄

**Comment by fr-animal** [comment]
LGTM!

**Pull request merged**